### PR TITLE
Add support for CJSX

### PR DESCRIPTION
**Pull request by zaro**
Use coffee-react-transform, to compile CJSX files also

### CODE REVIEW

**Comment by adrianlee44** [comment]
Thanks for the PR, @zaro.

It seems like the change will run all the code through the transform function. I was wondering what would happen when pipping vanilla coffeescript through that function.

**Comment by zaro** [comment]
Well CJSX is coffeescript with embedded JSX templates, so everything works just fine. 
But maybe I can make it an option in the settings, so that people not using CJSX don't run needless transforms?

**Comment by adrianlee44** [comment]
Thanks for clarifying that.

Adding an option and having it off by default would be great.

**Comment by zaro** [comment]
Added 'Compile CJSX' option which defaults to false. Also 'coffee-react-transform' module is not loaded at all when this option is disabled.

**Comment by zaro** [comment]
I see that the Travis build is failing, and I tried to look at the errors, and I cannot imagine how my change could possibly cause it :)

File: lib/main.js [hunk 1]
```diff
@@ -29,6 +29,10 @@ module.exports = {
     focusEditorAfterCompile: {
       type: 'boolean',
       "default": false
+    },
+    compileCjsx: {
+      type: 'boolean',
+      "default": false
```
Comment (adrianlee44) [inline lib/main.js]: Oh wow. I must have add this compiled js version into source last time. This isn't necessary but I'll remove this file in separate commit.

**Comment by adrianlee44** [comment]
Thanks for the PR. I'll merge this and will bump the version once I clean up the test errors.

**Comment by adrianlee44** [comment]
Coffee compile bumped to v0.12.0 with CJSX support

**Pull request merged**